Modules/mathmodule.c

```diff
@@ -769,7 +769,15 @@ long_lcm(PyObject *a, PyObject *b)
     if (_PyLong_IsZero((PyLongObject *)a) || _PyLong_IsZero((PyLongObject *)b)) {
         return PyLong_FromLong(0);
     }
-    g = _PyLong_GCD(a, b);
+
+    /* Make sure |a| <= |b| to speed up (a // g) * b; see gh-102221 for details. */
```

Comment:
This does not actually guarantee that `|a| <= |b|`.

Reply:
Yeah, it's actually about number of limbs, not absolute value.

```suggestion
    /* Make sure a_size <= b_size to speed up (a // g) * b; see gh-102221 for details. */
```